src/edge/tests/test_genome.py

```diff
@@ -383,3 +383,101 @@ def test_get_changed_locations_by_fragment(self):
                 )
             else:
                 raise Exception("Unexpected fragment")
+
+    def test_get_coordinate_diff_from_parent_genome(self):
+        genome = Genome.create("Foo")
+        self.assertEquals(len(genome.fragments.all()), 0)
+
+        # add initial sequence
+        s = "atggcatattcgcagct"
+        genome.add_fragment("chrI", s)
+        g1 = genome.indexed_genome()
+
+        # insert
+        g_u = g1.update()
+        with g_u.update_fragment_by_name("chrI") as f:
+            f.insert_bases(3, "gataca")
+            f.remove_bases(10, 4)
+        g2 = g_u.indexed_genome()
+
+        # insert again
+        g_u = g2.update()
+        with g_u.update_fragment_by_name("chrI") as f:
+            f.insert_bases(9, "gataca")
+        g3 = g_u.indexed_genome()
+
+        # insert again inside of previous integration
+        g_u = g3.update()
+        with g_u.update_fragment_by_name("chrI") as f:
+            f.insert_bases(5, "gataca")
+        g4 = g_u.indexed_genome()
+
+        diff1 = g2.get_coordinate_diff_from_parent_genome(g1)
+        truediff1 = [
+            {'parent_fragment_name': 'chrI', 'parent_fragment_id': 28,
+             'parent_start': 3, 'parent_end': 2,
```

Comment:
this is a bit confusing, where parent_start is > parent_end in case of an insertion. i know we've seen this type of logic before. perhaps we can use `parent_starts_at` and `parent_ends_before` as field names and then we would use python semantics, where the latter is the entity after ending?

Reply:
yeah I was wondering if I should change that, I believe this is how the UCSC/bed standard designates a region of length 0, since it's 1-based indexing, but I agree, it's just confusing. I'll switch over to use the python semantics for ending as you suggested.